packages/frontend/editor-ui/src/views/Evaluations.ee/utils.ts

```diff
@@ -13,30 +77,76 @@ export function getTestCasesColumns(
 		new Set([] as string[]),
 	);
 
-	return Array.from(inputColumnNames.keys()).map((column) => ({
-		prop: `${columnType}.${column}`,
+	return Array.from(inputColumnNames.keys());
+}
+
+function mapToColumns(
+	columnNames: string[],
+	columnType: 'inputs' | 'outputs' | 'metrics',
+	numeric?: boolean,
+): Column[] {
+	return columnNames.map((column) => ({
+		key: `${columnType}.${column}`,
 		label: column,
-		sortable: true,
-		filter: true,
-		showHeaderTooltip: true,
-		formatter: (row: TestCaseExecutionRecord) => {
-			const value = row[columnType]?.[column];
-			if (typeof value === 'object' && value !== null) {
-				return JSON.stringify(value, null, 2);
-			}
-
-			return `${value}`;
-		},
+		visible: true,
+		disabled: false,
+		columnType,
+		numeric,
 	}));
 }
 
-export function mapToNumericColumns(columnNames: string[]) {
-	return columnNames.map((metric) => ({
-		prop: `metrics.${metric}`,
-		label: metric,
-		sortable: true,
-		filter: true,
-		showHeaderTooltip: true,
-		formatter: (row: TestCaseExecutionRecord) => row.metrics?.[metric]?.toFixed(2) ?? '-',
-	}));
+function formatValue(
+	key: string,
+	value?: JsonValue,
+	{ numeric }: { numeric?: boolean } = { numeric: false },
+) {
+	let stringValue: string;
+
+	if (
+		numeric &&
+		typeof value === 'number' &&
+		!(key.endsWith('Tokens') || key === 'executionTime')
```

Comment:
For consistency we can also define special keys array in global scope and use `.includes` everywhere across this file (like on line 11)